src/browser-runner/platform-electron/sagas/server-command-handlers.js

```diff
@@ -28,6 +28,14 @@ function* createWindow({ meta: client, payload: { winId, url, width, height, sty
     }[style],
   });
 
+  win.on('close', (e) => {
+    // console.log('Window closing!!!', win.webContents);
+    // I'd like to do this only if there are no tabs left.
+    // Any ideas how I could get the number of tabs from here?
+    e.preventDefault();
+    return false;
+  });
```

Comment:
I *think* you just want to eat the `ctrl+w` event in electron, and also proceed as described above.